campaigns/tests.py

```
@@ -0,0 +1,317 @@
+        self.assertEqual(campaigns[0], campaign2)  # Newer campaign first
+        self.assertEqual(campaigns[1], campaign1)
+
+    def test_unique_on_chain_id(self):
+        """Test that on_chain_id must be unique"""
+        Campaign.objects.create(**self.campaign_data)
+
+        # Try to create another campaign with same on_chain_id
+        campaign_data_2 = self.campaign_data.copy()
+        campaign_data_2["name"] = "Duplicate Campaign"
+
+        with self.assertRaises(Exception):  # Should raise IntegrityError
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)